src/util/tryCatch.ts

```diff
@@ -11,7 +11,7 @@ function tryCatcher(): any {
   }
 }
 
-export function tryCatch(fn: Function): Function {
+export function tryCatch<T extends Function>(fn: T): T {
   tryCatchTarget = fn;
-  return tryCatcher;
-};
\ No newline at end of file
+  return <any>tryCatcher;
```

Comment:
Compiler returns `Type '() => any' is not assignable to type 'T'.` without it.

With `<T extends Function>` though we can see much nicer behavior when using the method.

Given

```
var predicate: (value: T, index: number, observable: Observable<T>) => boolean;
var tryCatchMethod: tryCatch(predicate);
```
The `typeof tryCatchMethod` will then be the same as `typeof predicate`.